pageserver/src/tenant/tasks.rs

```diff
@@ -65,6 +66,12 @@ pub(crate) async fn concurrent_background_tasks_rate_limit_permit(
         .with_label_values(&[loop_kind.as_static_str()])
         .guard();
 
+    fail_point!(
+        "initial-size-calculation-permit-pause",
```

Comment:
This should use the `pausable_failpoint!` macro from the parent module to prevent blocking the executor.

Reply:
Perhaps that's what I was missing. `delay` on the failpoint is used to unblock the executor. Thanks for the suggestion. I'll give it a go.